src/lotus/client.rs

```diff
@@ -95,6 +95,22 @@ impl<T: JsonRpcClient> LotusJsonRPCClient<T> {
     }
 }
 
+#[async_trait]
+impl<T: JsonRpcClient + Send + Sync> LotusBottomUpCheckpointClient for LotusJsonRPCClient<T> {
+    async fn ipc_has_voted_in_epoch(
```

Comment:
The Lotus side of this one has been implemented here: https://github.com/consensus-shipyard/lotus/pull/149